plugins/parsers/csv/parser.go

```diff
@@ -99,9 +168,17 @@ func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
 	return parseCSV(p, r)
 }
 
-// ParseLine does not use any information in header and assumes DataColumns is set
-// it will also not skip any rows
 func (p *Parser) ParseLine(line string) (telegraf.Metric, error) {
+	if len(line) == 0 {
+		if p.SkipRows > 0 {
```

Comment:
Maybe I can explain it @sspaink... We account for the case where you have a file and call `ParseLine` multiple times. For example, imaging a plugin (e.g. tail or file) to read the file line-by-line and call `ParseLine` each time a new line arrives. In this case you do have a state. You first should do nothing (i.e. skip) for the first `SkipRows` number of lines, later maybe read `N` header lines etc. Without reflecting this state in the parser it is hard to keep track of where you are (skipping, reading metadata or headers or data) in the process. Actually this is nothing invented by this PR, but it is already in the code.

We can now argue if it is a good idea to overwrite the user-specified config or if it would have been better to add another fields, but that's not the fault of this PR...

Reply:
I appreciate the explanation, I was confused about how SkipRows worked.

Reply:
That's a good point @srebhan:
>We can now argue if it is a good idea to overwrite the user-specified config or if it would have been better to add another fields, but that's not the fault of this PR...

What happens if telegraf config gets reloaded while still running? It will skip x rows again in that case?